Running YCSB from source checkout fails against 'basic' datastore binding.

Heya!
I tried this both on Windows and Linux. YCSB Compiles fine from original code, but when I try to run './bin/ycsb shell basic' the following errors will pop up in the shell:

[WARN]  Running against a source checkout. In order to get our runtime dependencies we'll have to invoke Maven. Depending on the state of your system, this may take ~30-45 seconds
[DEBUG]  Running 'mvn -pl com.yahoo.ycsb:basic-binding -am package -DskipTests dependency:build-classpath -DincludeScope=compile -Dmdep.outputFilterFile=true'
[ERROR]  Attempting to generate a classpath from Maven failed with return code '1'. The output from Maven follows, try running 'mvn -DskipTests package dependency:build=classpath' on your own and correct errors.

mvn output:
[INFO] Scanning for projects...
[ERROR] Could not find the selected project in the reactor: com.yahoo.ycsb:basic-binding -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MavenExecutionException


The mvn output appears to be generated when executing other ycsb operations as well although I have not tried many. YCSB without any parameters executes fine. I am not overly familiar with Maven, hence I have no idea what the problem is, or how to fix it. The relative consistency between Ubuntu and Windows, and the simplicity of the executed instructions may indicate it's a problem with Maven and/or YCSB rather than what I did. Anyway, any help is appreciated. I must build the project from code since I aim to implement new features I need for a rather specific test. For now, no modifications have been implemented, however.

Cheers,
Stefan Elsen

--- a/bin/ycsb
+++ b/bin/ycsb
@@ -138,11 +138,12 @@ def is_distribution():
 # presumes maven can run, so should only be run on source checkouts
 # will invoke the 'package' goal for the given binding in order to resolve intra-project deps
 # presumes maven properly handles system-specific path separators
-def get_classpath_from_maven(database):
+# Given module is full module name eg. 'core' or 'couchbase-binding'
+def get_classpath_from_maven(module):
     try:
-        debug("Running 'mvn -pl com.yahoo.ycsb:"+database+" -am package -DskipTests "
+        debug("Running 'mvn -pl com.yahoo.ycsb:" + module + " -am package -DskipTests "
               "dependency:build-classpath -DincludeScope=compile -Dmdep.outputFilterFile=true'")
-        mvn_output = subprocess.check_output(["mvn", "-pl", "com.yahoo.ycsb:" + database,
+        mvn_output = subprocess.check_output(["mvn", "-pl", "com.yahoo.ycsb:" + module,
                                               "-am", "package", "-DskipTests",
                                               "dependency:build-classpath",
                                               "-DincludeScope=compile",

--- a/bin/ycsb
+++ b/bin/ycsb
@@ -140,9 +140,9 @@ def is_distribution():
 # presumes maven properly handles system-specific path separators
 def get_classpath_from_maven(database):
     try:
-        debug("Running 'mvn -pl com.yahoo.ycsb:"+database+"-binding -am package -DskipTests "
+        debug("Running 'mvn -pl com.yahoo.ycsb:"+database+" -am package -DskipTests "
               "dependency:build-classpath -DincludeScope=compile -Dmdep.outputFilterFile=true'")
-        mvn_output = subprocess.check_output(["mvn", "-pl", "com.yahoo.ycsb:"+database+"-binding",
+        mvn_output = subprocess.check_output(["mvn", "-pl", "com.yahoo.ycsb:" + database,
                                               "-am", "package", "-DskipTests",
                                               "dependency:build-classpath",
                                               "-DincludeScope=compile",
@@ -200,12 +200,14 @@ def main():
         warn("Running against a source checkout. In order to get our runtime "
              "dependencies we'll have to invoke Maven. Depending on the state "
              "of your system, this may take ~30-45 seconds")
-        db_dir = os.path.join(ycsb_home, binding)
+        db_location = "core" if binding == "basic" else binding
+        project = "core" if binding == "basic" else binding + "-binding"
+        db_dir = os.path.join(ycsb_home, db_location)
         # goes first so we can rely on side-effect of package
-        maven_says = get_classpath_from_maven(binding)
+        maven_says = get_classpath_from_maven(project)
         # TODO when we have a version property, skip the glob
         cp = find_jars(os.path.join(db_dir, "target"),
-                       binding + "-binding*.jar")
+                       project + "*.jar")
         # alredy in jar:jar:jar form
         cp.append(maven_says)
     cp.insert(0, os.path.join(db_dir, "conf"))

